ddtrace/appsec/processor.py

```diff
@@ -0,0 +1,119 @@
+import errno
+import json
+import os.path
+from typing import Any
+from typing import ClassVar
+from typing import Optional
+from typing import TYPE_CHECKING
+
+import attr
+
+import ddtrace
+from ddtrace import config
+from ddtrace.appsec._ddwaf import DDWaf
+from ddtrace.constants import MANUAL_KEEP_KEY
+from ddtrace.internal.logger import get_logger
+from ddtrace.utils.formats import get_env
+
+
+if TYPE_CHECKING:
+    from ddtrace import Span
+
+ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
+DEFAULT_RULES = os.path.join(ROOT_DIR, "rules.json")
+
+log = get_logger(__name__)
+
+
+@attr.s(eq=False)
+class AppSecProcessor(object):
```

Comment:
I'm a bit uncertain if we want to follow the singleton pattern done in `RuntimeMetrics`.

The AppSecProcessor is coupled to a Tracer instance and so having global enable/disable methods will lock us into only using the global tracer.

I think making AppSecProcessor a SpanProcessor and enabling/disabling through the existing Tracer API (add as an option to `configure`, insert processor in `_initialize_span_processors`) would be a preferred design at the moment.

Reply:
Please check https://github.com/DataDog/dd-trace-py/pull/3175/files